Add v23.1.20 release notes (#18508)
cockroachdb/docs: src/current/_includes/releases/v23.1/v23.1.20.md

- Added three new [cluster settings]({% link v23.1/cluster-settings.md %}) for controlling [optimizer table statistics]({% link v23.1/cost-based-optimizer.md %}#table-statistics) forecasting:
    1. `sql.stats.forecasts.min_observations` is the minimum number of observed statistics required to produce a forecast.
    1. `sql.stats.forecasts.min_goodness_of_fit` is the minimum R² (goodness of fit) measurement required from all predictive models to use a forecast.
    1. `sql.stats.forecasts.max_decrease` is the most a prediction can decrease, expressed as the minimum ratio of the prediction to the lowest prior observation. [#123149][#123149]

<h3 id="v23-1-20-bug-fixes">Bug fixes</h3>

- Statistics forecasts of zero rows by [the optimizer]({% link v23.1/cost-based-optimizer.md %}#table-statistics) can cause bad plans. This commit changes forecasting to avoid predicting zero rows for most downward-trending statistics. [#123149][#123149]
- A [job]({% link v23.1/show-jobs.md %}) will now [log]({% link v23.1/logging.md %}#ops) rather than fail if it reports an out-of bound progress fraction. [#123133][#123133]

<h3 id="v23-1-20-performance-improvements">Performance improvements</h3>

> - Added a new [session variable]({% link v23.1/set-vars.md %}) `optimizer_use_improved_zigzag_join_costing`. When enabled, the cost of [zigzag joins]({% link v23.1/cost-based-optimizer.md %}#zigzag-joins) is updated so they will be never be chosen over scans unless they produce fewer rows. This change only matters if the session setting `enable_zigzag_join` is also `true`. [#123152][#123152]

Reviewer: ```suggestion
- Added a new [session variable]({% link v23.1/set-vars.md %}) `optimizer_use_improved_zigzag_join_costing`. When enabled, the cost of [zigzag joins]({% link v23.1/cost-based-optimizer.md %}#zigzag-joins) is updated so zigzag joins will only be chosen over scans if the zigzag joins produce fewer rows. This change only applies if the session setting `enable_zigzag_join` is also `true` (which is its initial value). [#123152][#123152]
```

My suggestion may sound repetitive, but I felt that "they" sounded ambiguous. I also noticed that `enable_zigzag_join`s  initial value is listed as `on` in the docs, but it seems that `true` is interchangeable with `on` for session variables.

Author: thanks for the suggestion, I think it's def clearer so I've updated it to repeat the term.  Also changed it to say `on` to match the docs and SQL shell

I have omitted the "which is its initial value" suggestion since this note can go out of date if that default were ever changed, and the canonical place that documents the default value of session vars is already linked to (and is better-maintained - if the default changes I doubt anyone will go back and catch this release note)
